### PR TITLE
API-1472: Add log when the event was not sent due to permissions on t…

### PR DESCRIPTION
**Pull request by pierrejolly**
**Description (for Contributor and Core Developer)**

Add log when the event was not sent because the connection does not have the required permissions

### CODE REVIEW

File: src/Akeneo/Connectivity/Connection/back/Application/Webhook/Service/EventPermissionLogger.php [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+use Akeneo\Platform\Component\EventQueue\EventInterface;
+
+/**
+ * @author Pierre Jolly <pierre.jolly@akeneo.com>
+ * @copyright 2021 Akeneo SAS (http://www.akeneo.com)
+ * @license http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+interface EventPermissionLogger
+{
+    public function logEventPermission(
```
Comment (tseho) [inline src/Akeneo/Connectivity/Connection/back/Application/Webhook/Service/EventPermissionLogger.php:16]: `logEventDataNotAccessible` ? what do you think ?
We cover both permissions & not found objects.

File: src/Akeneo/Connectivity/Connection/back/Application/Webhook/Service/EventPermissionLogger.php [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+        string $connectionCode,
+        EventInterface $event
+    ): void;
+}
```
[the remark this answers is not in the document]
Comment (pyaillet) [reply src/Akeneo/Connectivity/Connection/back/Application/Webhook/Service/EventPermissionLogger.php:20]: `EventsApiBuildErrorLogger` instead. WDYT ?

[the remark this answers is not in the document]
Comment (tseho) [reply src/Akeneo/Connectivity/Connection/back/Application/Webhook/Service/EventPermissionLogger.php:20]: `EventResourceNotAccessibleLogger` ?

[the remark this answers is not in the document]
Comment (pchasle) [reply src/Akeneo/Connectivity/Connection/back/Application/Webhook/Service/EventPermissionLogger.php]: > `EventsApiBuildErrorLogger` instead. WDYT ?
`ApiEventBuildErrorLogger` meaning that we are building an `ApiEvent`, whereas `EventsApi` is the feature, but both are still comprehensible

**Review by tseho — APPROVED** [review]

File: src/Akeneo/Connectivity/Connection/back/Application/Webhook/WebhookEventBuilder.php [hunk 3]
```diff
@@ -127,6 +141,8 @@ private function buildWebhookEvents(
             );
         }
 
+        $this->eventsApiDebugRepository->flush();
+
```
Comment (pchasle) [inline src/Akeneo/Connectivity/Connection/back/Application/Webhook/WebhookEventBuilder.php:145]:
```suggestion
```
You don't need to flush here, this is done in the Command Handler at the end of the event processing

**Review by pchasle — COMMENTED** [review]
You are missing a unit test (phpspec) in the `WebhookEventBuilder` to test that we log the error

File: src/Akeneo/Connectivity/Connection/back/Application/Webhook/WebhookEventBuilder.php (hunk 3 above)
Comment (pierrejolly) [reply src/Akeneo/Connectivity/Connection/back/Application/Webhook/WebhookEventBuilder.php:145]: Ok, so I don't need the eventsApiDebugRepository at all in this class 👌

**Review by tomglvng — APPROVED** [review]

**Pull request merged**